### PR TITLE
[WIP] Fei Savings Rate

### LINKED ISSUE #723: Fei Savings Rate

### PR DESCRIPTION
**Pull request by xklob**
[ more details to come later ]

### CODE REVIEW

File: contracts/fei/ibFei.sol [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+library IBFeiConfig {
+    address internal constant CORE_ADDRESS =
+        address(0x8d5ED43dCa8C2F7dFB20CF7b53CC7E593635d7b9);
+    address internal constant FEI_ADDRESS =
+        address(0x956F47F50A910163D8BF957Cf5846D573E7f87CA);
+    string internal constant IB_TOKEN_NAME = "ibFei";
+    string internal constant IB_TOKEN_SYMBOL = "ibFei";
```
[the remark this answers is not in the document]
Comment (ditchfieldcaleb) [reply contracts/fei/ibFei.sol:15]: Fwiw I've never heard of iron bank

[the remark this answers is not in the document]
Comment (ditchfieldcaleb) [reply contracts/fei/ibFei.sol:15]: But yes will capitalize.

File: contracts/fei/FSRX.sol [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import {ERC20} from "../external/solmate/tokens/ERC20.sol";
+import {ERC4626} from "../external/solmate/mixins/ERC4626.sol";
+import {TribeRoles} from "../core/TribeRoles.sol";
+import {CoreRef} from "../refs/CoreRef.sol";
+import {Constants} from "../Constants.sol";
+
+// SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (thomas-waite) [inline contracts/fei/FSRX.sol:7]: Tiny nit, would put the license identifier right at the top to be consistent

File: contracts/fei/FSRX.sol [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+pragma solidity ^0.8.4;
+
+library FSRXConfig {
```
Comment (thomas-waite) [inline contracts/fei/FSRX.sol:10]: Not sure the extra code of breaking out these constants into a library is worthwhile/makes things clearer. I'd just put inside the scope of the contract as constants

File: contracts/fei/FSRX.sol [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    address internal constant CORE_ADDRESS = address(0x8d5ED43dCa8C2F7dFB20CF7b53CC7E593635d7b9);
+    address internal constant FEI_ADDRESS = address(0x956F47F50A910163D8BF957Cf5846D573E7f87CA);
+
+    string internal constant TOKEN_NAME = "Fei Savings Rate";
+    string internal constant TOKEN_SYMBOL = "FSRX";
+
+    uint256 internal constant MAX_INTEREST_RATE = 1_000_000;
+
+    event InterestRateUpdate(uint256 newRate);
```
Comment (thomas-waite) [inline contracts/fei/FSRX.sol:19]: Why not include the old interest rate in this event? Would make it easier for clients/frontends so track what's going on. This is unlikely to be called often, so minimal gas overhead

File: contracts/fei/FSRX.sol (hunk 3 above)
Comment (ditchfieldcaleb) [reply contracts/fei/FSRX.sol:10]: I think it's much cleaner tbh

Comment (ElliotFriedman) [reply contracts/fei/FSRX.sol:10]: why not use CoreRef?

File: contracts/fei/FSRX.sol [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    /// @dev accrues interest by updating the conversion rate of FSRX to FEI
+    function accrueInterest() internal {
+        fsrxPrice = getConversionRate();
+        lastUpdated = block.timestamp;
+    }
+
+    /// @dev gets the current conversion rate of FSRX to FEI
+    function getConversionRate() public view returns (uint256) {
+        // Note: this is noncompounding. Need to change to continouosly compounding.
+        return fsrxPrice + (fsrxPrice * (block.timestamp - lastUpdated) * interestRate) / (10_000 * 365.25 days);
```
Comment (ElliotFriedman) [inline contracts/fei/FSRX.sol:88]: take a look at the compound codebase to see how to implement APY auto-compounding math

**Pull request closed without merging**